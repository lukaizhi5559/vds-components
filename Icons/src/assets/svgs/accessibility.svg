<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 25.3.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Ebene_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 21.6 21.6" style="enable-background:new 0 0 21.6 21.6;" xml:space="preserve">
<path d="M19.6,10.5c-1.9-3-5.3-4.7-8.8-4.7c-3.7,0-6.9,1.7-8.8,4.7l-0.2,0.3L2,11.1c1.9,3,5.2,4.7,8.8,4.7c3.6,0,6.9-1.7,8.8-4.7
	l0.2-0.3L19.6,10.5z M3.1,10.8c1-1.5,2.5-2.7,4.3-3.3c-1.7,1.8-1.7,4.7,0,6.5C5.6,13.5,4.1,12.3,3.1,10.8z M10.8,14.5
	c-2,0-3.7-1.6-3.7-3.7s1.6-3.7,3.7-3.7c2,0,3.7,1.6,3.7,3.7C14.5,12.8,12.8,14.5,10.8,14.5z M14.3,14.1c1.7-1.8,1.7-4.7,0-6.5
	c1.7,0.6,3.2,1.7,4.3,3.3C17.5,12.3,16,13.5,14.3,14.1z M12.3,10.8c0,0.8-0.7,1.5-1.5,1.5s-1.5-0.7-1.5-1.5s0.7-1.5,1.5-1.5l0,0
	C11.6,9.3,12.3,10,12.3,10.8z"/>
</svg>
