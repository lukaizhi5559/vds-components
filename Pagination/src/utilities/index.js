export { default as sharedDefaultProps } from './sharedDefaultProps';
export { default as sharedPropTypes } from './sharedPropTypes';
