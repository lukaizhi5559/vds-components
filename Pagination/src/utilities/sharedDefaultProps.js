export default {
  baseUrl: null,
  total: 20,
  selectPage: null,
  selectedPage: 10,
  surface: 'light',
  focusBorderRadius: '2px',
};
