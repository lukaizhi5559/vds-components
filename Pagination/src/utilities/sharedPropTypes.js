import React from 'react';
import PropTypes from 'prop-types';

export default {
  baseUrl: PropTypes.func,
  current: PropTypes.number,
  selectedPage: PropTypes.number,
  selectPage: PropTypes.func,
  total: PropTypes.number.isRequired,
  surface: PropTypes.oneOf(['light', 'dark']),
  focusBorderRadius: PropTypes.string,
};
