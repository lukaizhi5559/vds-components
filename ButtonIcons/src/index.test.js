import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ColorTokens } from '@vds-tokens/color';
import { AccessibilityTokens } from '@vds-tokens/accessibility';
import { calculateRem, hexToRgba } from '@vds-core/utilities';
import { ButtonIcon } from './index';
import { Icon } from '@vds-core/icons';

const focusringWidth = AccessibilityTokens.focusring.borderwidth.value;
const focusringStyle = AccessibilityTokens.focusring.borderstyle.value;
const focusRingLight = AccessibilityTokens.color.focusring.onlight.value;
const focusRingDark = AccessibilityTokens.color.focusring.ondark.value;
const grayPalleteToken = ColorTokens.palette.gray44.value;

describe('kind ghost/default - Light Surface', () => {
  test('default icon button background and icon style', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule('background-color', 'transparent');
    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe(
      ColorTokens.elements.primary.onlight.value
    );
  });
  test('icon button container size for fitToIcon prop - size large', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        fitToIcon={true}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule('width', '100%');
    expect(iconBtn).toHaveStyleRule('width', '100%');
  });
  test('icon button container size for fitToIcon prop - size small', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        fitToIcon={true}
        size="small"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn.parentNode).toHaveStyleRule('width', '1.25rem');
    expect(iconBtn.parentNode).toHaveStyleRule('height', '1.25rem');
    expect(iconBtn).toHaveStyleRule('width', '100%');
    expect(iconBtn).toHaveStyleRule('height', '100%');
  });
  test('icon button background styles on hover', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseOver(iconBtn);
    // border color with 6% opacity
    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      `0 0 0 0.0625rem ${hexToRgba(grayPalleteToken, 0.06)}`,
      {
        modifier: `:hover`,
      }
    );
    // background color with 6% opacity
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      `${hexToRgba(grayPalleteToken, 0.06)}`,
      {
        modifier: `:hover`,
      }
    );
  });
  test('icon color when icon button is active', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseDown(iconBtn);
    expect(iconBtn).toHaveStyleRule(
      'fill',
      `${ColorTokens.interactive.active.onlight.value}`,
      {
        modifier: `:active svg path`,
      }
    );
  });
  test('icon button styles on keyboard focus', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.keyDown(iconBtn, {
      key: 'Tab',
      keyCode: 9,
      charCode: 9,
    });
    expect(iconBtn).toHaveStyleRule(
      'border',
      `${calculateRem(focusringWidth)} ${focusringStyle}  ${focusRingLight}`,
      {
        modifier: `:focus:not(:hover)::before`,
      }
    );
  });
  test('icon color when icon button is disabled', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'light'}
        disabled={false}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe('#000000');
  });

  test('Button Icon selectable is selected', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'light'}
        disabled={false}
        selectable={true}
        selected={true}
        renderIcon={props => <Icon name="add-to-favorite" {...props} />}
        renderSelectedIcon={props => (
          <Icon name="added-to-favorite" {...props} />
        )}
      />
    );

    const iconBtnSvgPath = container.querySelector(
      '[class^="StyledLabel"] path'
    );
    expect(iconBtnSvgPath.getAttribute('fill')).toBe(
      `${ColorTokens.elements.brandhighlight.value}`
    );
  });

  test('Button Icon selectable custom svg unselected', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'light'}
        disabled={false}
        selectable={true}
        selected={false}
        renderIcon={props => (
          <svg
            role="img"
            width="20"
            aria-hidden="false"
            aria-label="add-to-favorite icon"
            viewBox="0 0 21.6 21.6"
          >
            <title>add-to-favorite icon</title>
            <g>
              <path
                d="M10.8,18.7l-7.2-6.9C2.4,10.6,1.8,9.4,1.8,8c0-2.8,2.3-5.1,5.1-5.1c1.6,0,3,0.7,4,1.9c1-1.2,2.4-1.9,3.9-1.9
		c2.8,0,5.1,2.3,5.1,5.1c0,1.8-1,3-2,4L10.8,18.7z M6.9,4c-2.2,0-4,1.8-4,3.9c0,1.1,0.4,2,1.4,3l6.5,6.2l6.3-6
		c1.1-1.1,1.6-2.1,1.6-3.2c0-2.2-1.8-3.9-4-3.9c-1.4,0-2.4,0.8-3.5,2l-0.4,0.5L10.4,6C9.4,4.7,8.3,4,6.9,4z"
                stroke="none"
                fill="#000000"
              ></path>
            </g>
          </svg>
        )}
        renderSelectedIcon={props => (
          <svg
            role="img"
            width="20"
            aria-hidden="false"
            aria-label="added-to-favorite icon"
            viewBox="0 0 21.6 21.6"
          >
            <title>added-to-favorite icon</title>
            <path
              d="M14.7,2.9c-1.5,0-2.9,0.7-3.9,1.9c-1-1.2-2.4-1.9-3.9-1.9C4.1,2.9,1.8,5.2,1.8,8c0,1.4,0.5,2.7,1.5,3.6l7.5,7.1l7.2-6.9
	c1.1-1,1.7-2.3,1.8-3.8C19.8,5.2,17.5,2.9,14.7,2.9z"
              stroke="none"
              fill="#000000"
            ></path>
          </svg>
        )}
      />
    );
    const iconfillcolor = container.querySelector('svg path');
    expect(iconfillcolor.getAttribute('fill')).toBe('#000000');
  });

  test('Button Icon selectable custom svg selected - surface light', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'light'}
        disabled={false}
        selectable={true}
        selected={true}
        renderIcon={props => (
          <svg
            role="img"
            width="20"
            aria-hidden="false"
            aria-label="add-to-favorite icon"
            viewBox="0 0 21.6 21.6"
          >
            <title>add-to-favorite icon</title>
            <g>
              <path
                d="M10.8,18.7l-7.2-6.9C2.4,10.6,1.8,9.4,1.8,8c0-2.8,2.3-5.1,5.1-5.1c1.6,0,3,0.7,4,1.9c1-1.2,2.4-1.9,3.9-1.9
		c2.8,0,5.1,2.3,5.1,5.1c0,1.8-1,3-2,4L10.8,18.7z M6.9,4c-2.2,0-4,1.8-4,3.9c0,1.1,0.4,2,1.4,3l6.5,6.2l6.3-6
		c1.1-1.1,1.6-2.1,1.6-3.2c0-2.2-1.8-3.9-4-3.9c-1.4,0-2.4,0.8-3.5,2l-0.4,0.5L10.4,6C9.4,4.7,8.3,4,6.9,4z"
                stroke="none"
                fill="#000000"
              ></path>
            </g>
          </svg>
        )}
        renderSelectedIcon={props => (
          <svg
            role="img"
            width="20"
            aria-hidden="false"
            aria-label="added-to-favorite icon"
            viewBox="0 0 21.6 21.6"
          >
            <title>added-to-favorite icon</title>
            <path
              d="M14.7,2.9c-1.5,0-2.9,0.7-3.9,1.9c-1-1.2-2.4-1.9-3.9-1.9C4.1,2.9,1.8,5.2,1.8,8c0,1.4,0.5,2.7,1.5,3.6l7.5,7.1l7.2-6.9
	c1.1-1,1.7-2.3,1.8-3.8C19.8,5.2,17.5,2.9,14.7,2.9z"
              stroke="none"
              fill="#000000"
            ></path>
          </svg>
        )}
      />
    );
    const iconBtnSvgPath = container.querySelector(
      '[class^="StyledLabel"] path'
    );
    expect(iconBtnSvgPath.getAttribute('fill')).toBe(`#000000`);
  });
});

describe('kind ghost/default - Dark Surface', () => {
  test('default icon button background and icon style', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule('background-color', 'transparent');
    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe(
      ColorTokens.elements.primary.ondark.value
    );
  });
  test('icon button background styles on hover', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseOver(iconBtn);
    // border color with 6% opacity
    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      `0 0 0 0.0625rem ${hexToRgba(grayPalleteToken, 0.26)}`,
      {
        modifier: `:hover`,
      }
    );
    // background color with 6% opacity
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      `${hexToRgba(grayPalleteToken, 0.26)}`,
      {
        modifier: `:hover`,
      }
    );
  });
  test('icon color when icon button is active', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseDown(iconBtn);
    expect(iconBtn).toHaveStyleRule(
      'fill',
      `${ColorTokens.interactive.active.ondark.value}`,
      {
        modifier: `:active svg path`,
      }
    );
  });
  test('icon button styles on keyboard focus', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.keyDown(iconBtn, {
      key: 'Tab',
      keyCode: 9,
      charCode: 9,
    });
    expect(iconBtn).toHaveStyleRule(
      'border',
      `${calculateRem(focusringWidth)} ${focusringStyle}  ${focusRingDark}`,
      {
        modifier: `:focus:not(:hover)::before`,
      }
    );
  });
  test('icon color when icon button is disabled', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'dark'}
        disabled={false}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe('#ffffff');
  });

  test('Button Icon selectable is selected', () => {
    const { container } = render(
      <ButtonIcon
        kind="ghost"
        surface={'dark'}
        disabled={false}
        selectable={true}
        selected={true}
        renderIcon={props => <Icon name="add-to-favorite" {...props} />}
        renderSelectedIcon={props => (
          <Icon name="added-to-favorite" {...props} />
        )}
      />
    );

    const iconBtnSvgPath = container.querySelector(
      '[class^="StyledLabel"] path'
    );
    expect(iconBtnSvgPath.getAttribute('fill')).toBe(
      `${ColorTokens.elements.primary.ondark.value}`
    );
  });
});

describe('Low Contrast - Light Surface - float false ', () => {
  test('Icon button default styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      hexToRgba(ColorTokens.palette.gray44.value, 0.06)
    );
  });

  test('Icon button hover styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseEnter(iconBtn);
    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      `0 0 0 ${calculateRem(1)} ${hexToRgba(
        ColorTokens.palette.gray44.value,
        0.06
      )}`,
      {
        modifier: `:hover`,
      }
    );
  });

  test('Icon button active styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );
    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.click(iconBtn);
    fireEvent.mouseDown(iconBtn);
    expect(iconBtn).toMatchSnapshot();
    expect(iconBtn).toHaveStyleRule(
      'fill',
      `${ColorTokens.interactive.active.onlight.value}`,
      {
        modifier: `:active svg path`,
      }
    );
  });

  test('Icon button keyboard focus styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.keyPress(iconBtn, {
      key: 'Tab',
      code: 9,
      charCode: 9,
    });
    expect(iconBtn).toHaveStyleRule('border', '0.0625rem dashed  #000000', {
      modifier: `:focus:not(:hover)::before`,
    });
  });

  test('Button Icon selectable is selected', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        surface={'light'}
        disabled={false}
        selectable={true}
        selected={true}
        renderIcon={props => <Icon name="add-to-favorite" {...props} />}
        renderSelectedIcon={props => (
          <Icon name="added-to-favorite" {...props} />
        )}
      />
    );

    const iconBtnSvgPath = container.querySelector(
      '[class^="StyledLabel"] path'
    );
    expect(iconBtnSvgPath.getAttribute('fill')).toBe(
      `${ColorTokens.elements.brandhighlight.value}`
    );
  });
});

describe('Low Contrast - Light Surface - float true ', () => {
  test('Icon button default styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        floating={true}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      ColorTokens.background.primary.light.value
    );
    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      '0px 1px 10px rgba(0,0,0,0.12),0px 2px 4px rgba(0,0,0,0.05)',
      {
        modifier: `:not(:hover)::after`,
      }
    );
  });

  test('Icon button hover styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        floating={true}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseEnter(iconBtn);

    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      `0 0 0 0.0625rem ${ColorTokens.elements.primary.ondark.value}`,
      {
        modifier: `:hover`,
      }
    );
    // expect(iconBtn).toHaveStyleRule(
    //   'border',
    //   `0.0625rem solid ${ColorTokens.elements.lowcontrast.onlight.value}`
    // );
  });

  test('Icon button keyboard focus styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        floating={true}
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.keyPress(iconBtn, {
      key: 'Tab',
      code: 9,
      charCode: 9,
    });
    expect(iconBtn).toHaveStyleRule('border', '0.0625rem dashed  #000000', {
      modifier: `:focus:not(:hover)::before`,
    });
  });
});

describe('Low Contrast - Dark Surface - float false ', () => {
  test('Icon button default styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      hexToRgba(ColorTokens.palette.gray44.value, 0.26)
    );
  });

  test('Icon button hover styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.mouseEnter(iconBtn);
    expect(iconBtn).toHaveStyleRule(
      'box-shadow',
      `0 0 0 ${calculateRem(1)} ${hexToRgba(
        ColorTokens.palette.gray44.value,
        0.26
      )}`,
      {
        modifier: `:hover`,
      }
    );
  });

  test('Icon button keyboard focus styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="lowContrast"
        surface="dark"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    fireEvent.keyPress(iconBtn, {
      key: 'Tab',
      code: 9,
      charCode: 9,
    });
    expect(iconBtn).toHaveStyleRule('border', '0.0625rem dashed  #ffffff', {
      modifier: `:focus:not(:hover)::before`,
    });
  });
});

describe('High Contrast - Light Surface ', () => {
  test('Icon button default styles', () => {
    const { container } = render(
      <ButtonIcon
        kind="highContrast"
        surface="light"
        renderIcon={props => <Icon name="app-dialer" {...props} />}
      />
    );

    const iconBtn = container.querySelector('[class^="StyledButton"]');
    expect(iconBtn).toHaveStyleRule(
      'background-color',
      ColorTokens.elements.primary.onlight.value
    );
    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe(
      ColorTokens.elements.primary.ondark.value
    );
  });

  test('Button Icon selectable is selected', () => {
    const { container } = render(
      <ButtonIcon
        kind="highContrast"
        surface={'light'}
        disabled={false}
        selectable={true}
        selected={true}
        renderIcon={props => <Icon name="add-to-favorite" {...props} />}
        renderSelectedIcon={props => (
          <Icon name="added-to-favorite" {...props} />
        )}
      />
    );

    const iconfillcolor = container.querySelector('[class^="StyledSVG"] path');
    expect(iconfillcolor.getAttribute('fill')).toBe(
      `${ColorTokens.elements.primary.ondark.value}`
    );
  });
});

describe('ButtonIcon - with BadgeIndicator', () => {
  test('render with badge indicator', () => {
    const { container } = render(
      <ButtonIcon
        kind="highContrast"
        surface={'light'}
        renderIcon={props => <Icon name="add-to-favorite" {...props} />}
        showBadgeIndicator
      />
    );

    expect(container).toMatchSnapshot();
  });
});
