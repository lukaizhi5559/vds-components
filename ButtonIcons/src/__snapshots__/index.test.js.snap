// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon - with BadgeIndicator render with badge indicator 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #ee0000;
  height: 1rem;
  min-width: 1rem;
  border-radius: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  background-color: #ffffff;
  height: 0.25rem;
  width: 0.25rem;
  border-radius: 0.25rem;
}

.c0 {
  position: relative;
  width: 2.75rem;
  height: 2.75rem;
}

.c1 {
  position: absolute;
  z-index: 1;
  height: 1rem;
  width: 1rem;
  left: calc(50% + 0.375rem);
  bottom: calc(50% + 0.375rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px;
  margin: 0px;
  cursor: pointer;
  box-sizing: border-box;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  touch-action: manipulation;
  pointer-events: auto;
  vertical-align: middle;
  outline: none;
  width: 100%;
  height: 100%;
  background-color: #000000;
  border-radius: 50%;
  border: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-transition: ease-out 0.1s;
  transition: ease-out 0.1s;
}

.c5:focus:not(:hover) {
  outline: none;
}

.c5:focus:not(:hover)::before {
  content: '';
  border: 0.0625rem dashed  #000000;
  border-radius: 50%;
  position: absolute;
  width: calc(100% + 0.25rem);
  height: calc(100% + 0.25rem);
}

.c5:focus:not(:hover):focus:not(:focus-visible)::before {
  border: none;
}

.c5:hover {
  outline: none;
  box-shadow: 0 0 0 0.0625rem #000000;
}

.c5:hover:active {
  box-shadow: 0 0 0 0.0625rem #6f7171;
}

.c5:active {
  background-color: #6f7171;
}

.c5:active svg path,
.c5:active svg polygon {
  fill: #ffffff;
}

.c5:focus:active:not(:hover) {
  box-shadow: none;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  bottom: 0;
  pointer-events: none;
}

.c7:focus {
  outline: none;
}

.c6 {
  height: 2.75rem;
  width: 2.75rem;
  display: inline-block;
  content: '';
  position: absolute;
  left: 50%;
  top: 50%;
  text-align: center;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c6:hover {
  cursor: pointer;
}

.c8 {
  display: inline-block;
  height: 1.25rem;
  width: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.c8:active,
.c8:focus {
  outline: none;
}

<div>
  <div
    class="c0"
    kind="highContrast"
  >
    <div
      class="c1"
      size="16"
    >
      <div
        class="c2"
      >
        <div
          aria-hidden="true"
          class="c3"
          kind="simple"
        >
          <div
            class="c4"
          />
        </div>
      </div>
    </div>
    <button
      aria-disabled="false"
      aria-hidden="false"
      aria-label="icon"
      class="c5"
      kind="highContrast"
      role="button"
      tabindex="0"
    >
      <span
        class="c6"
      />
      <span
        class="c7"
      >
        <svg
          aria-hidden="false"
          aria-label="add-to-favorite icon"
          class="c8"
          fill="#ffffff"
          role="img"
          viewBox="0 0 21.6 21.6"
        >
          <title>
            add-to-favorite icon
          </title>
          <g>
            <path
              d="M10.8,18.7l-7.2-6.9C2.4,10.6,1.8,9.4,1.8,8c0-2.8,2.3-5.1,5.1-5.1c1.6,0,3,0.7,4,1.9c1-1.2,2.4-1.9,3.9-1.9
		c2.8,0,5.1,2.3,5.1,5.1c0,1.8-1,3-2,4L10.8,18.7z M6.9,4c-2.2,0-4,1.8-4,3.9c0,1.1,0.4,2,1.4,3l6.5,6.2l6.3-6
		c1.1-1.1,1.6-2.1,1.6-3.2c0-2.2-1.8-3.9-4-3.9c-1.4,0-2.4,0.8-3.5,2l-0.4,0.5L10.4,6C9.4,4.7,8.3,4,6.9,4z"
              fill="#ffffff"
              stroke="none"
            />
          </g>
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Low Contrast - Light Surface - float false  Icon button active styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px;
  margin: 0px;
  cursor: pointer;
  box-sizing: border-box;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  touch-action: manipulation;
  pointer-events: auto;
  vertical-align: middle;
  outline: none;
  width: 100%;
  height: 100%;
  background-color: rgba(111,113,113,0.06);
  border-radius: 50%;
  border: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-transition: ease-out 0.1s;
  transition: ease-out 0.1s;
}

.c0:focus:not(:hover) {
  outline: none;
}

.c0:focus:not(:hover)::before {
  content: '';
  border: 0.0625rem dashed  #000000;
  border-radius: 50%;
  position: absolute;
  width: calc(100% + 0.25rem);
  height: calc(100% + 0.25rem);
}

.c0:focus:not(:hover):focus:not(:focus-visible)::before {
  border: none;
}

.c0:hover {
  outline: none;
  box-shadow: 0 0 0 0.0625rem rgba(111,113,113,0.06);
}

.c0:hover:active {
  box-shadow: 0 0 0 0.0625rem rgba(111,113,113,0.06);
}

.c0:active svg path,
.c0:active svg polygon {
  fill: #6f7171;
}

.c0:focus:active:not(:hover) {
  box-shadow: none;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  bottom: 0;
  pointer-events: none;
}

.c2:focus {
  outline: none;
}

.c1 {
  height: 2.75rem;
  width: 2.75rem;
  display: inline-block;
  content: '';
  position: absolute;
  left: 50%;
  top: 50%;
  text-align: center;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1:hover {
  cursor: pointer;
}

.c3 {
  display: inline-block;
  height: 1.25rem;
  width: 1.25rem;
  min-height: 1.25rem;
  min-width: 1.25rem;
}

.c3:active,
.c3:focus {
  outline: none;
}

<button
  aria-disabled="false"
  aria-hidden="false"
  aria-label="icon"
  class="c0"
  kind="lowContrast"
  role="button"
  tabindex="0"
>
  <span
    class="c1"
  />
  <span
    class="c2"
  >
    <svg
      aria-hidden="false"
      aria-label="app-dialer icon"
      class="c3"
      fill="#000000"
      role="img"
      viewBox="0 0 21.6 21.6"
    >
      <title>
        app-dialer icon
      </title>
      <path
        d="M6.70574,6.71227H1.8V1.8H6.70574ZM2.91688,5.59412h2.672v-2.676h-2.672Zm10.3292,1.11815H8.33937V1.8h4.90671ZM9.45625,5.59412h2.673v-2.676H9.45625ZM19.78546,6.71227H14.87875V1.8h4.90671ZM15.99563,5.59412h2.67295v-2.676H15.99563Zm-9.27535,7.662H1.81454V8.34387H6.72028ZM2.93142,12.138h2.672V9.462h-2.672Zm10.32921,1.11814H8.35392V8.34387h4.90671ZM9.4708,12.138h2.673V9.462H9.4708ZM19.8,13.25613H14.89329V8.34387H19.8ZM16.01017,12.138h2.673V9.462H16.01017ZM6.70574,19.8H1.8V14.88773H6.70574ZM2.91688,18.68185h2.672v-2.676h-2.672ZM13.24608,19.8H8.33937V14.88773h4.90671ZM9.45625,18.68185h2.673v-2.676H9.45625ZM19.78546,19.8H14.87875V14.88773h4.90671Zm-3.78983-1.11815h2.67295v-2.676H15.99563Z"
        fill="#000000"
        stroke="none"
      />
    </svg>
  </span>
</button>
`;
